specs/dwritecore/DWriteCore.md

```diff
@@ -0,0 +1,255 @@
+font cache or persisted cache file. Second, a restricted factory only enumerates well-known fonts in the system
+font collection. The well-known-fonts restriction enables DWriteCore to efficiently construct the system font 
+collection without depending on a cross-process or persistent cache. An example use case for a restricted factory
+would be a web browser rendering process with very limited permissions.
+
+A _bitmap render target_ (IDWriteBitmapRenderTarget) is an API object that encapsulates a system memory bitmap
+and enables rendering glyphs to the bitmap. On Windows, the object encapsulates a GDI memory DC with a GDI
+device-dependent bitmap (DIB) selected into it. However, on non-Windows platforms, there is no GDI DC,
+and the bitmap is just a system memory array. DWriteCore introduces a new version of the bitmap render target
+interface that an application can use to access the bitmap data without going through GDI.
+
+# Examples
+
+## Creating a restricted factory
+
+An application calls the [DWriteCreateFactory](https://docs.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-dwritecreatefactory)
+function to create a factory object, which is the entry-point to all other DirectWrite APIs. An application can
+create a _restricted factory_ if it does not want the factory or objects created from it to read from or write
+to any cross-process or persistent cache. In addition, a restricted factory's system font collection only includes
+well-known system fonts.
+
+The following example creates a restricted factory.
+
+```c++
+ComPtr<IDWriteFactory7> spFactory;
+HRESULT hr = DWriteCreateFactory(
+    DWRITE_FACTORY_TYPE_RESTRICTED,
+    __uuidof(IDWriteFactory7),
+    (IUnknown**)&spFactory
+    );
+```
+
+## Drawing glyphs to a system memory bitmap
+
+An application creates a bitmap render target by calling [IDWriteGdiInterop::CreateBitmapRenderTarget](https://docs.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-idwritegdiinterop-createbitmaprendertarget).
+A bitmap render target encapsulates a bitmap in system memory, and eanbles rendering glyphs to the bitmap. On Windows,
+the bitmap is actually a GDI DIB selected into a memory HDC, and it is possible to get the bitmap pixels using GDI
+functions. However, DWriteCore introduces a simpler method through the new IDWriteBitmapRenderTarget2 interface.
+The following example shows both the old way and the new way.
+
+```c++
+DWRITE_BITMAP_DATA_BGRA32 TextRenderer::GetBitmapData(IDWriteBitmapRenderTarget* renderTarget)
+{
+    DWRITE_BITMAP_DATA_BGRA32 result = {};
+
+    // Query the bitmap render target for the new interface.
+    ComPtr<IDWriteBitmapRenderTarget2> renderTarget2;
+    HRESULT hr = renderTarget->QueryInterface(&renderTarget2);
+    if (SUCCEEDED(hr))
+    {
+        // IDWriteBitmapRenderTarget2 exists, so we can get the bitmap the easy way.
+        hr = renderTarget2->GetBitmapData(OUT &result);
```

Comment:
Please don't abuse SAL annotations like this.

Reply:
RECOMMEND: remove the `OUT` keyword, if you are compelled use `/* out */` as an inline comment.